internal/provider/resource_gitlab_system_hook.go

```diff
@@ -0,0 +1,179 @@
+				ForceNew:    true,
+			},
+			"merge_requests_events": {
+				Description: "Trigger hook on merge requests events.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"repository_update_events": {
+				Description: "Trigger hook on repository update events.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"enable_ssl_verification": {
+				Description: "Do SSL verification when triggering the hook.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				ForceNew:    true,
+			},
+			"created_at": {
+				Description: "The date and time the hook was created in ISO8601 format.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+})
+
+func resourceGitlabSystemHookCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	options := &gitlab.AddHookOptions{
+		URL: gitlab.String(d.Get("url").(string)),
+	}
+	// NOTE: `GetOkExists()` is deprecated, but until there is a replacement we need to use it.
+	//       see https://github.com/hashicorp/terraform-plugin-sdk/pull/350#issuecomment-597888969
+
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("token"); ok {
+		options.Token = gitlab.String(v.(string))
+	}
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("push_events"); ok {
+		options.PushEvents = gitlab.Bool(v.(bool))
+	}
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("tag_push_events"); ok {
+		options.TagPushEvents = gitlab.Bool(v.(bool))
+	}
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("merge_requests_events"); ok {
+		options.MergeRequestsEvents = gitlab.Bool(v.(bool))
+	}
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("repository_update_events"); ok {
+		options.RepositoryUpdateEvents = gitlab.Bool(v.(bool))
+	}
+	// nolint:staticcheck // SA1019 ignore deprecated GetOkExists
+	// lintignore: XR001 // TODO: replace with alternative for GetOkExists
+	if v, ok := d.GetOkExists("enable_ssl_verification"); ok {
+		options.EnableSSLVerification = gitlab.Bool(v.(bool))
+	}
+
+	log.Printf("[DEBUG] create gitlab system hook %q", *options.URL)
+
+	hook, _, err := client.SystemHooks.AddHook(options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%d", hook.ID))
+	if options.Token != nil {
+		d.Set("token", options.Token)
+	}
```

Comment:
This shouldn't be needed. After all you set `options.Token` by getting the `"token"` attribute already.
```suggestion
```